lib/lattice_observer/observed/event_processor.ex

```diff
@@ -229,35 +249,75 @@ defmodule LatticeObserver.Observed.EventProcessor do
 
     l = record_host(l, source_host, labels, stamp, friendly_name)
 
+    # legacy heartbeat has a list for the actors field...
+    # default to "new format" if this field is missing
     l =
-      List.foldl(Map.get(data, "actors", []), l, fn x, acc ->
-        put_actor_instance(
-          acc,
-          source_host,
-          x["public_key"],
-          x["instance_id"],
-          spec,
-          stamp,
-          %{}
-        )
-      end)
+      if is_list(Map.get(data, "actors", %{})) do
+        put_legacy_instances(l, source_host, spec, stamp, data)
+      else
+        actors_expanded =
+          Enum.flat_map(Map.get(data, "actors", %{}), fn {k, count} ->
+            Enum.map(1..count, fn _ -> k end)
+          end)
 
+        l =
+          List.foldl(actors_expanded, l, fn pk, acc ->
+            put_actor_instance(
+              acc,
+              source_host,
+              pk,
+              "n/a",
+              spec,
+              stamp,
+              %{}
+            )
+          end)
+
+        List.foldl(Map.get(data, "providers", []), l, fn x, acc ->
+          put_provider_instance(
+            acc,
+            source_host,
+            x["public_key"],
+            x["link_name"],
+            "n/a",
```

Comment:
This `n/a` should be the `x["contract_id"]`

Reply:
there's no contract ID on the new heartbeat